### PR TITLE
libpng: restore generation of FindPNG.cmake by CMakeDeps & modernize more

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libpng/all/conanfile.py [hunk 1]
```diff
@@ -72,48 +71,24 @@ def config_options(self):
...
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("zlib/1.2.13")
 
+    def validate(self):
+        if Version(self.version) < "1.6" and self.info.settings.arch == "armv8" and is_apple_os(self):
+            raise ConanInvalidConfiguration(f"{self.ref} could not be cross build on Mac.")
```
Comment (jwillikers) [inline recipes/libpng/all/conanfile.py:86]: Can this be worded slightly better? This doesn't actually have to do with cross-compiling does it? It looks like this doesn't support building for `armv8` for Apple systems, whether or not that's native right?

**Review by jwillikers — APPROVED** [review]

File: recipes/libpng/all/conanfile.py (hunk 1 above)
Comment (prince-chrismc) [reply recipes/libpng/all/conanfile.py:86]: I'd suggest a move positive language here too

```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} currently does not support building with {self.info.settings.arch}. Please contribute.")
```

**Review by prince-chrismc — COMMENTED** [review]
Just the one nit - otherwise its good

File: recipes/libpng/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/libpng/all/conanfile.py:86]: I would say "does not support building for"

Comment (prince-chrismc) [reply recipes/libpng/all/conanfile.py:86]:
```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} currently does not building for {self.info.settings.os} {self.info.settings.arch}. Contributions are welcomed")
```

**Review by prince-chrismc — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**